state/whitelist.js

```diff
@@ -0,0 +1,35 @@
+import { createContainer } from "unstated-next";
+import { useEffect, useState } from "react";
+import { getAddress } from "ethers/lib/utils";
+import axios from "axios";
+import { WVET } from "vexchange-sdk";
+
+function useWhitelist() {
+    // Addresses are in checksummed format
+    const [whiteListedTokens, setWhitelistedTokens] = useState(null);
+    const FOUNDATION_WHITELIST_URL = "https://vechain.github.io/token-registry/main.json";
+
+    // --> Lifecycle: on mount
+    useEffect(() => {
+        const fetchWhitelist = async () => {
+            const result = await axios.get(FOUNDATION_WHITELIST_URL);
+            const whitelist = {};
+            result.data.forEach((token) => {
+                // Converts all addresses into checksummed addresses
+                const tokenContractAddress = getAddress(token.address);
+                whitelist[tokenContractAddress] = token;
+            })
+            // Need to include the WVET address manually as it is not in the foundation whitelist
+            whitelist[WVET[1].address] = WVET[1];
+            setWhitelistedTokens(whitelist);
```

Comment:
Is there a good reason to prefer an object over a map? Like for logging the object or something? In general, I find the map API far cleaner and more modern (although a lot of the benefits come in TypeScript specifically).

Reply:
You're right. It's better especially in typescript. So I changed the implementation from object to a `Map` instead